nowcasting_dataset/dataset/datasets.py

```diff
@@ -314,3 +364,91 @@ def worker_init_fn(worker_id):
         # The NowcastingDataset copy in this worker process.
         dataset_obj = worker_info.dataset
         dataset_obj.per_worker_init(worker_info.id)
+
+
+def subselect_data(
+    batch: example.Example,
+    required_keys: Union[Tuple[str], List[str]],
+    current_timestep_index: int,
+    history_minutes: int,
+    forecast_minutes: int,
+) -> example.Example:
+    """
+    Subselects the data temporally. This function selects all data within the range [history_minutes, forecast_minutes]
+
+    Args:
+        batch: Example dictionary containing at least the required_keys
+        required_keys: The required keys present in the dictionary to use
+        current_timestep_index: The index into either SATELLITE_DATETIME_INDEX or NWP_TARGET_TIME giving the current timestep
+        history_minutes: How many minutes of history to use
+        forecast_minutes: How many minutes of future data to use for forecasting
+
+    Returns:
+        Example with only data between [history_minutes, forecast_minutes] remaining
+    """
+
+    # We are subsetting the data
+    date_time_index_to_use = (
+        SATELLITE_DATETIME_INDEX if SATELLITE_DATA in required_keys else NWP_TARGET_TIME
+    )
+    current_time = batch[date_time_index_to_use].isel(time=current_timestep_index)[0]
+    # Datetimes are in seconds, so just need to convert minutes to second + 30sec buffer
+    # Only need to do it for the first example in the batch, as masking indicies should be the same for all of them
+    # The extra 30 seconds is added to ensure that there to ensure that the first and last timestep are always contained
```

Comment:
Thanks for adding an explanation!  A tiny typo fix:

```suggestion
    # The extra 30 seconds is added to ensure that the first and last timestep are always contained
```